### PR TITLE
fix: allow multiple user messages in a row for google gemini

### LINKED ISSUE #258: allow multiple user messages in a row for google gemini
Google does not allow multiple same-role messages in a row and it throws `Please ensure that multiturn requests ends with a user role or a function response.` error message.

This issue was created from this PR: https://github.com/Portkey-AI/gateway/pull/246 .

### PR DESCRIPTION
**Pull request by meronogbai**
**Title:** 
- This will allow users to include multiple user messages in a row in gemini-pro.

**Description:** (optional)
- Inspired by [this code](https://github.com/PublicAffairs/openai-gemini/blob/d16cfafa897364151dc4a4eac6d901e9b8c97e45/src/worker.mjs#L202-L215)

**Motivation:** (optional)
- An app I work on relies on multiple user messages and it was breaking for me when I tried to use it with gemini-pro.

**How to test?**

The following snippet breaks with gemini-pro even though it works with openai. Checkout this branch and everything should work without issues 🫡

```ts
async function main() {
  const stream = await openai.chat.completions.create({
    model: "gemini-pro",
    messages: ([
      {
        role: "user",
        content: "Say this is a test",
      },
      { role: "user", content: "now say im happy" },
    ]),
    stream: true,
    max_tokens: 4096,
  });

  for await (const chunk of stream) {
    process.stdout.write(chunk.choices[0]?.delta?.content || "");
  }
}

main();
```

### CODE REVIEW

File: src/providers/google/chatComplete.ts [hunk 1]
```diff
@@ -64,7 +67,17 @@ export const GoogleChatCompleteConfig: ProviderConfig = {
                         }
                     });
                 }
-                return { role, parts };
+
+                // @NOTE: This takes care of the "Please ensure that multiturn requests alternate between user and model."
+                // error that occurs when we have multiple user messages in a row.
+                const shouldAppendEmptyModeChat = lastRole === 'user' && role === 'user' && !params.model?.includes('vision');
```
Comment (meronogbai) [inline src/providers/google/chatComplete.ts:73]: Opting out in case of `gemini-pro-vision` because it doesn't support multiturn chat anyway.

**Comment by vrushankportkey** [comment]
Thank you so much for the PR, @meronogbai! This is amazing. Paging @VisargD who can check this

**Comment by VisargD** [comment]
Thanks for the PR @meronogbai . I will review and merge this PR today.

**Comment by VisargD** [comment]
LGTM. Closes #258

**Review by VisargD — APPROVED** [review]

**Pull request merged**